dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/airflow/AirflowConnector.java

```diff
@@ -95,24 +117,123 @@ public void addTasksTo(@Nonnull List<? super Task<?>> out, @Nonnull ConnectorArg
+    //     "task_instance",
+    //     "select * from task_instance ",
+    //     zonedIntervals,
+    //     (startDate, endDate) ->
+    //         String.format(
+    //             " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as
+    // TIMESTAMP) ;",
+    //             dateToSqlFormat(startDate), dateToSqlFormat(endDate)));
+    // // Airflow v1.6.0
+    // addTableTaskWithIntervals(
+    //     out,
+    //     "dag_run",
+    //     "select * from dag_run ",
+    //     zonedIntervals,
+    //     (startDate, endDate) ->
+    //         String.format(
+    //             " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as
+    // TIMESTAMP) ;",
+    //             dateToSqlFormat(startDate), dateToSqlFormat(endDate)));
+
+    addQueryTask(
+        out,
+        "task_instance.csv",
+        dateRange == null
+            ? "select * from task_instance;"
+            : "select * from task_instance "
+                + String.format(
+                    " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as TIMESTAMP) ;",
+                    dateToSqlFormat(dateRange.getLeft()), dateToSqlFormat(dateRange.getRight())));
+
     // Airflow v1.6.0
-    addFullTable(out, "dag_run.csv", "select * from dag_run;");
+    addQueryTask(
+        out,
+        "dag_run.csv",
+        dateRange == null
+            ? "select * from dag_run;"
+            : "select * from dag_run "
+                + String.format(
+                    " where end_date >= CAST( '%s' as TIMESTAMP) and end_date < CAST( '%s' as TIMESTAMP) ;",
+                    dateToSqlFormat(dateRange.getLeft()), dateToSqlFormat(dateRange.getRight())));
 
     // Airflow v1.10.7
     // analog of DAG's python definition in json
-    addFullTable(out, "serialized_dag.csv", "select * from serialized_dag;", TaskCategory.OPTIONAL);
+    addQueryTask(out, "serialized_dag.csv", "select * from serialized_dag;", TaskCategory.OPTIONAL);
     // Airflow v1.10.10
-    addFullTable(out, "dag_code.csv", "select * from dag_code;", TaskCategory.OPTIONAL);
+    addQueryTask(out, "dag_code.csv", "select * from dag_code;", TaskCategory.OPTIONAL);
+  }
+
+  @Nullable
+  private ZonedIntervalIterable fromAirflowConnectorArguments(ConnectorArguments arguments) {
+    Pair<ZonedDateTime, ZonedDateTime> dateRange = dateRange(arguments);
+    if (dateRange == null) {
+      LOG.info("Date ranges was not specified. Generate full table queries.");
+      return null;
+    }
+
+    LOG.info(
+        "Date range for query generation from {} to {} exclusive and increments of one day.",
+        dateRange.getLeft(),
+        dateRange.getRight());
+    return ZonedIntervalIterableGenerator.forDateRangeWithIntervalDuration(
+        dateRange.getLeft(), dateRange.getRight(), Duration.ofDays(1));
+  }
+
+  @Nullable
+  private Pair<ZonedDateTime, ZonedDateTime> dateRange(ConnectorArguments arguments) {
+    boolean isDateRangeSpecified =
```

Comment:
nit: I think `isDateRangeSpecified` is redundant. You can immediately return if (arguments.getStartDate() == null || arguments.getLookbackDays() == null)

Reply:
That right, but there is a technic of self documented code. The variable is introduced to describe the boolean expression instead of writing a comment.